pkg/server/commands/listusers/list_users_rpc_test.go

```diff
@@ -1507,7 +1507,7 @@ func TestListUsersExclusion(t *testing.T) {
 				tuple.NewTupleKey("group:1", "blocked", "group:1#member"),
```

Comment:
Can we add this test as well please? It makes sure that if we experience cycles through independent object_ids and with a non-stratifiable resolution on the `document:1#viewer` and `document:2#blocked` relationship then we still yield the residual relationships for `user:jon`.

```go
		{
			name: "exclusion_with_chained_negation",
			req: &openfgav1.ListUsersRequest{
				Object:   &openfgav1.Object{Type: "document", Id: "2"},
				Relation: "viewer",
				UserFilters: []*openfgav1.UserTypeFilter{
					{
						Type: "user",
					},
				},
			},
			model: `model
			  schema 1.1

			type user

			type document
			  relations
			    define unblocked: [user]
				define blocked: [user, document#viewer] but not unblocked
				define viewer: [user, document#blocked] but not blocked
			`,
			tuples: []*openfgav1.TupleKey{
				tuple.NewTupleKey("document:1", "viewer", "document:2#blocked"),
				tuple.NewTupleKey("document:2", "blocked", "document:1#viewer"),
				tuple.NewTupleKey("document:2", "viewer", "user:jon"),
				tuple.NewTupleKey("document:2", "unblocked", "user:jon"),
			},
			expectedUsers: []string{"user:jon"},
		},
```

Reply:
Please add this test as well, and it is currently failing. This is a tricky corner case involving non-stratifiable queries through exclusion.

```
{
	name: "non_stratifiable_exclusion_containing_cycle",
	req: &openfgav1.ListUsersRequest{
		Object:   &openfgav1.Object{Type: "document", Id: "1"},
		Relation: "viewer",
		UserFilters: []*openfgav1.UserTypeFilter{
			{
				Type:     "document",
				Relation: "blocked",
			},
		},
	},
	model: `model
		schema 1.1

	type user

	type document
		relations
		define blocked: [user, document#viewer]
		define viewer: [user, document#blocked] but not blocked
	`,
	tuples: []*openfgav1.TupleKey{
		tuple.NewTupleKey("document:1", "viewer", "document:2#blocked"),
		tuple.NewTupleKey("document:2", "blocked", "document:1#viewer"),
	},
	expectedUsers: []string{},
},
```

For the same `Check(document:1#viewer@document:2#blocked)` this should return `{allowed: false}` as it is a non-stratifable query.

Reply:
This is a good point. The missing piece was handling the base-cycle case. There are zero situations of negation where a cycle can return a result in ListUsers.